### PR TITLE
add approval modal

### PR DESCRIPTION
**Pull request by XiaoYhun**
image
image

### CODE REVIEW

File: src/pages/SwapV2/index.tsx [hunk 1]
```diff
@@ -861,22 +862,30 @@ export default function Swap() {
...
-                                </AutoRow>
+                                </Row>
                               ) : approvalSubmitted && approval === ApprovalState.APPROVED ? (
                                 <Trans>Approved</Trans>
                               ) : (
-                                <Trans>Approve ${currencyIn?.symbol}</Trans>
+                                <Row justify="center" gap="6px">
+                                  <InfoHelper
+                                    color={theme.textReverse}
+                                    text={t`You need to first allow KyberSwap's smart contract to use your KNC`}
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV2/index.tsx:883]: KNC

File: src/pages/SwapV3/ApprovalModal.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+      e.preventDefault()
+      setOption(prev => (prev === ApproveOptions.Infinite ? ApproveOptions.Custom : ApproveOptions.Infinite))
+    }
+  }
+  return (
+    <Modal isOpen={isOpen} onDismiss={closeModal}>
+      <Wrapper onKeyDown={handleKeyDown}>
+        <RowBetween marginBottom="16px">
+          <Text fontSize="20px" lineHeight="24px">
+            Approve
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/ApprovalModal.tsx:132]: trans

File: src/constants/permit.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    '0xfa1FBb8Ef55A4855E5688C0eE13aC3f202486286': { type: PermitType.AMOUNT, version: '1' }, //FHM
+    '0x9879aBDea01a879644185341F7aF7d8343556B7a': { type: PermitType.AMOUNT, version: '1' }, //TUSD
+    '0x6496994241804D7fE2b032901931e03bCD82301F': { type: PermitType.AMOUNT, version: '1' }, //MODA
+    '0xB66b5D38E183De42F21e92aBcAF3c712dd5d6286': { type: PermitType.AMOUNT, version: '1' }, //PILLS
+    '0xfB98B335551a418cD0737375a2ea0ded62Ea213b': { type: PermitType.AMOUNT, version: '1' }, //miMATIC
+    '0xDDc0385169797937066bBd8EF409b5B3c0dFEB52': { type: PermitType.AMOUNT, version: '1' }, //wMEMO
+  },
+}
+
+export const EIP712_DOMAIN_TYPE = [
```
Comment (nguyenhoaidanh) [inline src/constants/permit.ts:160]: duplicate `EIP712Domain`

File: src/pages/SwapV3/ApprovalModal.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+                target="_blank"
+                rel="noreferrer"
+              >
+                Read more ↗
+              </a>
+            </Trans>
+          </Text>
+          <OptionWrapper active={option === ApproveOptions.Infinite} onClick={() => setOption(ApproveOptions.Infinite)}>
+            <RowFit flex="0 0 48px">
+              <svg width="44" height="44" viewBox="0 0 44 44" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (viet-nv) [inline src/pages/SwapV3/ApprovalModal.tsx:154]: Can we move this svg to an asset file?

File: src/pages/SwapV3/ApprovalModal.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                    fill="currentcolor"
+                  />
+                </g>
+                <g clipPath="url(#clip1_398_6043)" filter="blur(2px)">
+                  <path
+                    d="M5.5 31.6249V38.4999H12.375L32.6517 18.2233L25.7767 11.3483L5.5 31.6249ZM37.9683 12.9066C38.6833 12.1916 38.6833 11.0366 37.9683 10.3216L33.6783 6.03161C32.9633 5.31661 31.8083 5.31661 31.0933 6.03161L27.7383 9.38661L34.6133 16.2616L37.9683 12.9066Z"
+                    fill="currentcolor"
+                  />
+                </g>
+              </svg>
```
Comment (viet-nv) [inline src/pages/SwapV3/ApprovalModal.tsx:213]: Can we use react feather?
https://feathericons.com/?query=edit

File: src/pages/SwapV2/index.tsx (hunk 1 above)
Comment (XiaoYhun) [reply src/pages/SwapV2/index.tsx:883]: are we still support swap v2 ?

Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx:883]: yes, only solana

File: src/constants/permit.ts (hunk 3 above)
Comment (XiaoYhun) [reply src/constants/permit.ts:160]: permit use many type of EIP712 domain, so better leave it a version here

Comment (nguyenhoaidanh) [reply src/constants/permit.ts:160]: EIP712Domain vs EIP712_DOMAIN_TYPE are the same. Let's remove 1 of them

File: src/pages/SwapV3/ApprovalModal.tsx (hunk 5 above)
Comment (XiaoYhun) [reply src/pages/SwapV3/ApprovalModal.tsx:213]: its different one, there is needed to blur the svg 
image

**Review by viet-nv — APPROVED** [review]

**Pull request merged**